test/models/links_test.rb

```diff
@@ -69,6 +69,19 @@ class LinksTest < ActiveSupport::TestCase
     assert_equal([["home", "https://example.code"], ["download", "/downloads/.gem"]], enumerated)
   end
 
+  should "always include source code even when duplicate" do
+    metadata = { "homepage_uri" => "https://example.com", "source_code_uri" => "https://example.com" }
```

Comment:
```suggestion
    metadata = { "homepage_uri" => "https://example.code", "source_code_uri" => "https://example.code" }
```

I think you mean this?